### ISSUE TITLE
Crash on r107 changing the parameter value of a function

### ISSUE THREAD

**Issue by Kyatric** [Scirra/Construct-3-bugs#1737]
## Problem description

Add or change the value of a parameter to a function. On validation, C3 crashes

## Attach a .c3p

Reproduced in new blank project

## Steps to reproduce

1. Start new project
2. Add function object
3. In the event sheet, create a on start of layout event
4. Add and action "Call function"
5. Name "Test"
5. Add a parameter 
6. Click "Done" (or press enter"
7. Crash

## Observed result

Crash

## Expected result

Change the value, no crash

## Affected browsers

* Chrome: yes
* Firefox: not tested
* Edge: not tested
* Safari: not tested

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Error report information
Type: unhandled rejection
Reason: Error: unexpected type @ TypeError: unexpected type at Function.ǃi.ǃJ (https://editor.construct.net/r107/main.js:2:110345) at new window.ǃeXq (https://editor.construct.net/r107/main.js:63:369139) at Function.ǃi.ǃN (https://editor.construct.net/r107/main.js:2:141255) at window.ǃeXt.Set (https://editor.construct.net/r107/main.js:63:373714) at ǃi.ǃevo.then.e (https://editor.construct.net/r107/components/editors/eventSheetView/eventSheetView.js:1:75595)
Stack: TypeError: unexpected type at Function.ǃi.ǃJ (https://editor.construct.net/r107/main.js:2:110345) at new window.ǃeXq (https://editor.construct.net/r107/main.js:63:369139) at Function.ǃi.ǃN (https://editor.construct.net/r107/main.js:2:141255) at window.ǃeXt.Set (https://editor.construct.net/r107/main.js:63:373714) at ǃi.ǃevo.then.e (https://editor.construct.net/r107/components/editors/eventSheetView/eventSheetView.js:1:75595)
Construct 3 version: r107
URL: https://editor.construct.net/r107/
Date: Tue Jun 26 2018 14:59:22 GMT+0200 (heure d’été d’Europe centrale)
Uptime: 55.6 s

Platform information
Browser: Chrome
Browser version: 67.0.3396.87
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 7
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36
C3 release: r107 (beta)
Language setting: en-US

WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 1050 Ti Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions: EXT_color_buffer_float, EXT_texture_filter_anisotropic, OES_texture_float_linear, WEBGL_compressed_texture_s3tc, WEBGL_compressed_texture_s3tc_srgb, WEBGL_debug_renderer_info, WEBGL_debug_shaders, WEBGL_lose_context
</details>

**Comment by AshleyScirra** [comment]
Ah, it only happens in the production version, not our development copy. That suggests some kind of issue with the build... which automatically makes this a difficult one :-\

It does not reproduce in r106, which is weird, I don't think we changed any of the code involved in this at all for r107...

**Comment by Kyatric** [comment]
I've modified the steps to give a bit more precisions.
This is consistent. I've reversed to r106 to keep on my production.

I also encounter a layout view issue, closing tabs, but I'm not sure I can reproduce, and I do suspect indeed something like a typo in the build/parameters of the build or something that is messing up stuff that otherwise do work indeed.

**Comment by shortercode** [comment]
@Kyatric what is the nature of the layout view issue? This crash appears to be a bug in the software that does the build, @AshleyScirra is working around it at the moment. The layout view issue is possibly unconnected, but the bug could be causing more issues than we're aware of.

**Comment by Kyatric** [comment]
@shortercode Well, apparently, in r107 with my project that has 3 ranks of tabs, I just have to open the project and click on the close icon of a tab to have C3 crashing.

I'll make an upload of the c3p (big one, it will take a bit of time) and send you a message for you to check if you need. 

I quickly tried in a new project, simply adding a few layouts doesn't do the trick.

**Comment by shortercode** [comment]
mmm yeah I tried opening up demonoire and a dozen layouts. Closing the tabs doesn't seem to cause a crash. Must be some step or project specific item that causes the issue.

I'm going to open a new issue for this bug, so we don't clutter this one up too much.

**Comment by AshleyScirra** [comment]
I think this is caused by the minifier mangling code badly (yet again) ☹️ I suspect each case is a variant of this report I've just filed: https://github.com/babel/minify/issues/875

These are really tough issues to fix. Essentially the minifier is garbling a few bits of code in random places and causes random bugs to appear, only after we've released a build. It doesn't happen in our development copies since they haven't been minified, debugging a minified build is a nightmare, and there are no easy or good workarounds. I think we might be reduced to shotgun changes and hoping it works again on the next build... but then we can't guarantee it hasn't garbled something in some other feature elsewhere in the editor so we might then just get different bug reports... this is going to be a nightmare

**Comment by AshleyScirra** [comment]
Rolling back the minifier a few versions seems to work around this. @Kyatric can you test with r107.2 and verify if it's working again?

**Comment by Kyatric** [comment]
@AshleyScirra The parameter I was trying to modify can be modified without crash.

**Comment by AshleyScirra** [comment]
Great, thanks for confirming. Hopefully this version of the minifier will avoid any further problems.
